ggml-opencl.cpp

```diff
@@ -1835,7 +1835,10 @@ static void ggml_cl_mul_mat_q_f32(const ggml_tensor * src0, const ggml_tensor *
                     CL_CHECK(clEnqueueNDRangeKernel(queue, *to_fp32_cl, 1, &offset, &global, local > 0 ? &local : NULL, events.size(), !events.empty() ? events.data() : NULL, NULL));
                 }
 
-                for (int64_t i12 = i02 * r2, e12 = i12 + r2; i12 < e12; i12++) {
+                int64_t i12 = i02 * r2;
+                int64_t e12 = i12 + r2;
+                events.reserve(e12 - i12);
+                while (i12 < e12) {
```

Comment:
Better to keep the `for` loop

Reply:
fix it https://github.com/ggerganov/llama.cpp/pull/7272/commits/4ee29e5e1caf29e1bc7b094226faa890ae0e98d6